src/OpenLoco/src/GameCommands/LowerLand.cpp

```diff
@@ -0,0 +1,99 @@
+#include "Audio/Audio.h"
+#include "Economy/Expenditures.h"
+#include "GameCommands.h"
+#include "Localisation/FormatArguments.hpp"
+#include "Localisation/StringIds.h"
+#include "Map/RoadElement.h"
+#include "Map/SurfaceData.h"
+#include "Map/SurfaceElement.h"
+#include "Map/TileLoop.hpp"
+#include "Map/TileManager.h"
+#include "Map/TrackElement.h"
+#include "Objects/LandObject.h"
+#include "Objects/ObjectManager.h"
+#include "S5/S5.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::World;
+using namespace OpenLoco::World::SurfaceSlope;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004638C6, 0x004638CF
+    static uint32_t lowerLand(const LowerLandArgs& args, std::set<World::Pos3, LessThanPos3>& removedBuildings, const uint8_t flags)
+    {
+        if (flags & Flags::apply)
+        {
+            S5::getOptions().madeAnyChanges = 1;
+
+            if (getCommandNestLevel() == 1 && getUpdatingCompanyId() != CompanyId::neutral)
+            {
+                const auto height = TileManager::getHeight(args.pointA).landHeight;
+                Audio::playSound(Audio::SoundId::construct, World::Pos3(args.pointA.x, args.pointA.y, height));
+            }
+        }
+
+        World::TilePosRangeView tileLoop{ toTileSpace(args.pointA), toTileSpace(args.pointB) };
+
+        // Find out what the highest baseZ in the selected range is
+        auto highestBaseZ = 0;
+        for (const auto& tilePos : tileLoop)
+        {
+            auto tile = TileManager::get(tilePos);
+            auto* surface = tile.surface();
+
+            auto baseZ = TileManager::getSurfaceCornerHeight(surface);
+            highestBaseZ = std::max<SmallZ>(highestBaseZ, baseZ);
+        }
+
+        // Now modify only the elements matching this highest baseZ
+        auto totalCost = 0;
+        for (const auto& tilePos : tileLoop)
+        {
+            auto tile = TileManager::get(tilePos);
+            auto* surface = tile.surface();
+
+            auto baseZ = TileManager::getSurfaceCornerHeight(surface);
+            if (baseZ < highestBaseZ)
+            {
+                continue;
+            }
+
+            auto targetBaseZ = highestBaseZ;
+            auto slopeFlags = lowerSurfaceCornerFlags(args.corner, surface->slope());
+            if (slopeFlags & SurfaceSlope::requiresHeightAdjustment)
+            {
+                targetBaseZ -= kSmallZStep;
+                slopeFlags &= ~SurfaceSlope::requiresHeightAdjustment;
+            }
```

Comment:
I believe this is matching what vanilla is doing, yet lowering slopes results in raising them. Not sure where this is going wrong.